### PR TITLE
SOLR-14920, SOLR-16029: contrib/ltr/src/java: apply & enforce 'spotless' code formatting

### PR DESCRIPTION
**Pull request by cpoerschke**
https://issues.apache.org/jira/browse/SOLR-14920

`src/java` code only in this pull request, `src/test` code to follow separately after this pull request is reviewed and merged, to keep review sizes manageable-ish hopefully.

### CODE REVIEW

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java [hunk 1]
```diff
@@ -19,15 +19,18 @@
...
 
 /**
- * FeatureLogger can be registered in a model and provide a strategy for logging
- * the feature values.
+ * FeatureLogger can be registered in a model and provide a strategy for logging the feature values.
  */
 public abstract class FeatureLogger {
 
-  /** the name of the cache using for storing the feature value **/
+  /** the name of the cache using for storing the feature value * */
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java:26]: this change surprises me.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java [hunk 2]
```diff
@@ -50,50 +49,52 @@
...
   // contains a description of the model
-  final private LTRScoringModel ltrScoringModel;
-  final private boolean extractAllFeatures;
-  final private LTRThreadModule ltrThreadMgr;
-  final private Semaphore querySemaphore; // limits the number of threads per query, so that multiple requests can be serviced simultaneously
+  private final LTRScoringModel ltrScoringModel;
+  private final boolean extractAllFeatures;
+  private final LTRThreadModule ltrThreadMgr;
+  private final Semaphore
+      querySemaphore; // limits the number of threads per query, so that multiple requests can be
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java:65]: perhaps moving the comment to a separate line would be clearer here.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java [hunk 3]
```diff
@@ -209,96 +209,108 @@ public ModelWeight createWeight(IndexSearcher searcher, ScoreMode scoreMode, flo
...
       for (final Feature.FeatureWeight fw : featureWeights) {
         extractedFeatureWeights[i++] = fw;
       }
-      for (final Feature f : modelFeatures){
-        modelFeaturesWeights[j++] = extractedFeatureWeights[f.getIndex()]; // we can lookup by featureid because all features will be extracted when this.extractAllFeatures is set
+      for (final Feature f : modelFeatures) {
+        modelFeaturesWeights[j++] =
+            extractedFeatureWeights[
+                f.getIndex()]; // we can lookup by featureid because all features will be
+        // extracted when this.extractAllFeatures is set
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java:234]: perhaps moving the comment to a separate line would be clearer here too.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java [hunk 4]
```diff
@@ -209,96 +209,108 @@ public ModelWeight createWeight(IndexSearcher searcher, ScoreMode scoreMode, flo
...
+    try {
       for (final Feature f : features) {
         CreateWeightCallable callable = new CreateWeightCallable(f, searcher, needsScores, req);
         RunnableFuture<Feature.FeatureWeight> runnableFuture = new FutureTask<>(callable);
-        querySemaphore.acquire(); // always acquire before the ltrSemaphore is acquired, to guarantee a that the current query is within the limit for max. threads
-        ltrThreadMgr.acquireLTRSemaphore();//may block and/or interrupt
-        ltrThreadMgr.execute(runnableFuture);//releases semaphore when done
+        querySemaphore
+            .acquire(); // always acquire before the ltrSemaphore is acquired, to guarantee a that
+        // the current query is within the limit for max. threads
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java:308]: perhaps moving the comment to a separate line would be clearer here too.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java [hunk 5]
```diff
@@ -420,16 +437,16 @@ private void setFeaturesInfo(){
...
+     * features that will be used for scoring. Then calculate and return the model's score.
      */
     private float makeNormalizedFeaturesAndScore() {
       int pos = 0;
       for (final Feature.FeatureWeight feature : modelFeatureWeights) {
         final int featureId = feature.getIndex();
         FeatureInfo fInfo = featuresInfo[featureId];
-        if (fInfo.isUsed()) { // not checking for finfo == null as that would be a bug we should catch
+        if (fInfo
+            .isUsed()) { // not checking for finfo == null as that would be a bug we should catch
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java:449]: perhaps moving the comment to a separate line would be clearer here too.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java [hunk 6]
```diff
@@ -460,23 +476,24 @@ public Explanation explain(LeafReaderContext context, int doc)
...
 
     protected void reset() {
-      for (int i = 0; i < extractedFeatureWeights.length;++i){
+      for (int i = 0; i < extractedFeatureWeights.length; ++i) {
         int featId = extractedFeatureWeights[i].getIndex();
         float value = extractedFeatureWeights[i].getDefaultValue();
-        featuresInfo[featId].setValue(value); // need to set default value everytime as the default value is used in 'dense' mode even if used=false
+        featuresInfo[featId].setValue(
+            value); // need to set default value everytime as the default value is used in 'dense'
+        // mode even if used=false
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRScoringQuery.java:487]: perhaps moving the comment to a separate line would be clearer here too.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRThreadModule.java [hunk 7]
```diff
@@ -28,35 +27,38 @@
...
+ * &lt;queryParser name="ltr" class="org.apache.solr.ltr.search.LTRQParserPlugin"&gt;
+ * &lt;int name="threadModule.totalPoolThreads"&gt;10&lt;/int&gt;
+ * &lt;int name="threadModule.numThreadsPerRequest"&gt;5&lt;/int&gt;
+ * &lt;/queryParser&gt;
+ *
+ * &lt;transformer name="features" class="org.apache.solr.ltr.response.transform.LTRFeatureLoggerTransformerFactory"&gt;
+ * &lt;int name="threadModule.totalPoolThreads"&gt;10&lt;/int&gt;
+ * &lt;int name="threadModule.numThreadsPerRequest"&gt;5&lt;/int&gt;
+ * &lt;/transformer&gt;
+ * </pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRThreadModule.java:47]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/Feature.java [hunk 8]
```diff
@@ -37,63 +36,67 @@
...
- * <p>
- * {@link Feature} is an abstract class and concrete classes should implement
- * the {@link #validate()} function, and must implement the {@link #paramsToMap()}
- * and createWeight() methods.
+ * "class" : "...",
+ * "name" : "myFeature",
+ * "params" : {
+ * ...
+ * }
+ * }</pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/Feature.java:56]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/FieldLengthFeature.java [hunk 9]
```diff
@@ -28,20 +27,23 @@
...
- * Note: since this feature relies on norms values that are stored in a single byte
- * the value of the feature could have a lightly different value.
- * (see also {@link org.apache.lucene.search.similarities.ClassicSimilarity})
- **/
+ * "name":  "titleLength",
+ * "class": "org.apache.solr.ltr.feature.FieldLengthFeature",
+ * "params": {
+ * "field": "title"
+ * }
+ * }</pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/FieldLengthFeature.java:41]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/FieldValueFeature.java [hunk 10]
```diff
@@ -41,27 +40,29 @@
...
-  "params": {
-      "field": "hits"
-  }
-}</pre>
+ * "name":  "rawHits",
+ * "class": "org.apache.solr.ltr.feature.FieldValueFeature",
+ * "params": {
+ * "field": "hits"
+ * }
+ * }</pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/FieldValueFeature.java:52]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/OriginalScoreFeature.java [hunk 11]
```diff
@@ -28,48 +27,56 @@
...
  * <pre>{
-  "name":  "originalScore",
-  "class": "org.apache.solr.ltr.feature.OriginalScoreFeature",
-  "params": { }
-}</pre>
- **/
+ * "name":  "originalScore",
+ * "class": "org.apache.solr.ltr.feature.OriginalScoreFeature",
+ * "params": { }
+ * }</pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/OriginalScoreFeature.java:39]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/SolrFeature.java [hunk 12]
```diff
@@ -41,22 +40,23 @@
...
+ * "class": "org.apache.solr.ltr.feature.SolrFeature",
+ * "params":{ "fq": ["{!terms f=category}book"] }
+ * },
+ * {
+ * "name":  "documentRecency",
+ * "class": "org.apache.solr.ltr.feature.SolrFeature",
+ * "params": {
+ * "q": "{!func}recip( ms(NOW,publish_date), 3.16e-11, 1, 1)"
+ * }
+ * }]</pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/SolrFeature.java:58]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/ValueFeature.java [hunk 13]
```diff
@@ -19,31 +19,31 @@
...
+ *
  * <pre>{
-   "name" : "userFromMobile",
-   "class" : "org.apache.solr.ltr.feature.ValueFeature",
-   "params" : { "value" : "${userFromMobile}", "required":true }
- }</pre>
+ * "name" : "userFromMobile",
+ * "class" : "org.apache.solr.ltr.feature.ValueFeature",
+ * "params" : { "value" : "${userFromMobile}", "required":true }
+ * }</pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/ValueFeature.java:37]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/interleaving/algorithms/TeamDraftInterleaving.java [hunk 14]
```diff
@@ -22,31 +22,31 @@
...
+ * Interleaving produces a fair distribution of ranking models’ elements in the final interleaved
+ * list. "Team draft interleaving" has also proved to overcome an issue of the "Balanced
+ * interleaving" approach, in determining the winning model[4].
+ *
+ * <p>[1] T. Joachims. Optimizing search engines using clickthrough data. KDD (2002) [2]
+ * T.Joachims.Evaluatingretrievalperformanceusingclickthroughdata.InJ.Franke, G. Nakhaeizadeh, and
+ * I. Renz, editors, Text Mining, pages 79–96. Physica/Springer (2003) [3] F. Radlinski, M. Kurup,
+ * and T. Joachims. How does clickthrough data reflect re- trieval quality? In CIKM, pages 43–52.
+ * ACM Press (2008) [4] O. Chapelle, T. Joachims, F. Radlinski, and Y. Yue. Large-scale validation
+ * and analysis of interleaved search evaluation. ACM TOIS, 30(1):1–41, Feb. (2012)
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/interleaving/algorithms/TeamDraftInterleaving.java:42]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/interleaving/algorithms/TeamDraftInterleaving.java [hunk 15]
```diff
@@ -57,20 +57,18 @@
...
+   * algorithm creates a unique ranked list I = (i1, i2, ...). This list is created by interleaving
+   * elements from the two lists la and lb as described by Chapelle et al.[1]. Each element Ij is
+   * labelled TeamA if it is selected from La and TeamB if it is selected from Lb.
+   *
+   * <p>[1] O. Chapelle, T. Joachims, F. Radlinski, and Y. Yue. Large-scale validation and analysis
+   * of interleaved search evaluation. ACM TOIS, 30(1):1–41, Feb. (2012)
+   *
+   * <p>Assumptions: - rerankedA and rerankedB has the same length. They contains the same search
+   * results, ranked differently by two ranking models - each reranked list can not contain the same
+   * search result more than once. - results are all from the same shard
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/interleaving/algorithms/TeamDraftInterleaving.java:71]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/DefaultWrapperModel.java [hunk 16]
```diff
@@ -35,32 +34,38 @@
...
-  "params": {
-    "resource": "models/myModel.json"
-  }
-}</pre>
+ * "class": "org.apache.solr.ltr.model.DefaultWrapperModel",
+ * "name": "myWrapperModelName",
+ * "params": {
+ * "resource": "models/myModel.json"
+ * }
+ * }</pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/DefaultWrapperModel.java:50]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java [hunk 17]
```diff
@@ -38,71 +37,81 @@
...
+ *   <li>a list of features ({@link Feature}) and
+ *   <li>a list of normalizers ({@link Normalizer}) plus
+ *   <li>parameters or configuration to represent the scoring algorithm.
  * </ul>
- * <p>
- * Example configuration (snippet):
+ *
+ * <p>Example configuration (snippet):
+ *
  * <pre>{
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java:51]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LinearModel.java [hunk 18]
```diff
@@ -19,79 +19,85 @@
...
- * A scoring model that computes scores using a dot product.
- * Example models are RankSVM and Pranking.
- * <p>
- * Example configuration:
+ * A scoring model that computes scores using a dot product. Example models are RankSVM and
+ * Pranking.
+ *
+ * <p>Example configuration:
+ *
  * <pre>{
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LinearModel.java:33]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/MultipleAdditiveTreesModel.java [hunk 19]
```diff
@@ -21,99 +21,102 @@
...
-   }
-}</pre>
- * <p>
- * Training libraries:
+ * A scoring model that computes scores based on the summation of multiple weighted trees. Example
+ * models are LambdaMART and Gradient Boosted Regression Trees (GBRT) .
+ *
+ * <p>Example configuration:
+ *
+ * <pre>{
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/MultipleAdditiveTreesModel.java:36]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/NeuralNetworkModel.java [hunk 20]
```diff
@@ -28,75 +27,79 @@
 
 /**
  * A scoring model that computes document scores using a neural network.
- * <p>
- * Supported <a href="https://en.wikipedia.org/wiki/Activation_function">activation functions</a> are:
- * <code>identity</code>, <code>relu</code>, <code>sigmoid</code>, <code>tanh</code>, <code>leakyrelu</code> and
- * contributions to support additional activation functions are welcome.
- * <p>
- * Example configuration:
-<pre>{
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/NeuralNetworkModel.java:37]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/WrapperModel.java [hunk 21]
```diff
@@ -29,26 +28,26 @@
...
+ * WrapperModel#fetchModelMap()}.
  *
- * <p>This model doesn't hold the actual parameters of the wrapped model,
- * thus it can manage large models which are difficult to upload to ZooKeeper.
+ * <p>This model doesn't hold the actual parameters of the wrapped model, thus it can manage large
+ * models which are difficult to upload to ZooKeeper.
  *
  * <p>Example configuration:
+ *
  * <pre>{
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/WrapperModel.java:40]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/norm/MinMaxNormalizer.java [hunk 22]
```diff
@@ -20,19 +20,22 @@
...
-<pre>
-"norm" : {
-    "class" : "org.apache.solr.ltr.norm.MinMaxNormalizer",
-    "params" : { "min":"0", "max":"50" }
-}
-</pre>
+ *
+ * <p>Example configuration:
+ *
+ * <pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/norm/MinMaxNormalizer.java:26]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/norm/StandardNormalizer.java [hunk 23]
```diff
@@ -20,20 +20,22 @@
...
-    "class" : "org.apache.solr.ltr.norm.StandardNormalizer",
-    "params" : { "avg":"42", "std":"6" }
-}
-</pre>
- * <p>
- * Example normalizations:
+ *
+ * <p>Example configuration:
+ *
+ * <pre>
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/norm/StandardNormalizer.java:26]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/package-info.java [hunk 24]
```diff
@@ -16,30 +16,24 @@
...
- * into the framework  by extending {@link org.apache.solr.ltr.model.LTRScoringModel},
- * (see for example {@link org.apache.solr.ltr.model.MultipleAdditiveTreesModel} and {@link org.apache.solr.ltr.model.LinearModel}).
- * </p>
- * <p>
- * The {@link org.apache.solr.ltr.LTRScoringQuery} will take care of computing the values of
- * all the features (see {@link org.apache.solr.ltr.feature.Feature}) and then will delegate the final score
- * generation to the {@link org.apache.solr.ltr.model.LTRScoringModel}, by calling the method
- * {@link org.apache.solr.ltr.model.LTRScoringModel#score(float[] modelFeatureValuesNormalized) score(float[] modelFeatureValuesNormalized)}.
- * </p>
+ * This package contains the main logic for performing the reranking using a Learning to Rank model.
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/package-info.java:19]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java [hunk 25]
```diff
@@ -51,18 +50,19 @@
 import org.apache.solr.util.SolrPluginUtils;
 
 /**
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java:52]: note to self: check this is still rendered correctly

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java [hunk 26]
```diff
@@ -162,31 +161,31 @@ private FeatureLogger createFeatureLogger(String formatStr) {
...
 
     private List<LeafReaderContext> leafContexts;
     private SolrIndexSearcher searcher;
     /**
-     * rerankingQueries, modelWeights have:
-     * length=1 - [Classic LTR] When reranking with a single model
-     * length=2 - [Interleaving] When reranking with interleaving (two ranking models are involved)
+     * rerankingQueries, modelWeights have: length=1 - [Classic LTR] When reranking with a single
+     * model length=2 - [Interleaving] When reranking with interleaving (two ranking models are
+     * involved)
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java:174]: this comment auto-reformatting impairs content clarity.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java [hunk 27]
```diff
@@ -210,99 +209,120 @@ public void setContext(ResultContext context) {
...
+     *
      * @param transformerFeatureStore the explicit transformer feature store
      */
     private LoggingModel createLoggingModel(String transformerFeatureStore) {
       final ManagedFeatureStore fr = ManagedFeatureStore.getManagedFeatureStore(req.getCore());
 
       final FeatureStore store = fr.getFeatureStore(transformerFeatureStore);
-      transformerFeatureStore = store.getName(); // if transformerFeatureStore was null before this gets actual name
+      transformerFeatureStore =
+          store.getName(); // if transformerFeatureStore was null before this gets actual name
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java:248]: perhaps moving the comment to a separate line would be clearer here too.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java [hunk 28]
```diff
@@ -210,99 +209,120 @@ public void setContext(ResultContext context) {
...
+     *
+     * <p>No Reranking There is the need of a logger model from the default feature store or the
+     * explicit feature store passed to extract the feature vector
+     *
+     * <p>Re Ranking 1) If no explicit feature store is passed, the models for each reranking query
+     * can be safely re-used the feature vector can be fetched from the feature vector cache. 2) If
+     * an explicit feature store is passed, and no reranking query uses a model with that feature
+     * store, There is the need of a logger model to extract the feature vector 3) If an explicit
+     * feature store is passed, and there is a reranking query that uses a model with that feature
+     * store, the model can be re-used and there is no need for a logging model
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java:264]: this comment auto-reformatting impairs content clarity.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/store/FeatureStore.java [hunk 29]
```diff
@@ -20,16 +20,16 @@
...
 
 public class FeatureStore {
 
-  /** the name of the default feature store **/
+  /** the name of the default feature store * */
   public static final String DEFAULT_FEATURE_STORE_NAME = "_DEFAULT_";
 
-  private final LinkedHashMap<String,Feature> store = new LinkedHashMap<>(); // LinkedHashMap because we need predictable iteration order
+  private final LinkedHashMap<String, Feature> store =
+      new LinkedHashMap<>(); // LinkedHashMap because we need predictable iteration order
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/store/FeatureStore.java:32]: perhaps moving the comment to a separate line would be clearer here too.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/store/rest/ManagedFeatureStore.java [hunk 30]
```diff
@@ -170,41 +162,41 @@ public void doGet(BaseSolrResource endpoint, String childId) {
...
       features.add(m);
     }
     return features;
   }
 
-  private static LinkedHashMap<String,Object> toFeatureMap(Feature feat) {
-    final LinkedHashMap<String,Object> o = new LinkedHashMap<>(4, 1.0f); // 1 extra for caller to add store
+  private static LinkedHashMap<String, Object> toFeatureMap(Feature feat) {
+    final LinkedHashMap<String, Object> o =
+        new LinkedHashMap<>(4, 1.0f); // 1 extra for caller to add store
```
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/store/rest/ManagedFeatureStore.java:185]: perhaps moving the comment to a separate line would be clearer here too.

**Review by cpoerschke — COMMENTED** [review]
Done human review for the automatic changes done by the `gradlew :solr:contrib:ltr:spotlessApply` command:
* some comment auto-reformatting was terrible and needs manual edits somehow to avoid it
* some comment relocation was unhelpful and could benefit from manual relocation
* the effect of the reformatting on javadoc and specifically the `<pre>...</pre>` markup use needs to be checked i.e. does it still render correctly?

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java (hunk 1 above)
Comment (madrob) [reply solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java:26]: I would manually change it to `*/`

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java [hunk 31]
```diff
@@ -36,46 +39,42 @@ protected FeatureLogger(String fvCacheName, FeatureFormat f) {
...
+      LTRScoringQuery.FeatureInfo[] featuresInfo) {
     final String featureVector = makeFeatureVector(featuresInfo);
     if (featureVector == null) {
       return false;
     }
 
-    return searcher.cacheInsert(fvCacheName,
-        fvCacheKey(scoringQuery, docid), featureVector) != null;
+    return searcher.cacheInsert(fvCacheName, fvCacheKey(scoringQuery, docid), featureVector)
+        != null;
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java:61]: I don't like this

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java [hunk 32]
```diff
@@ -34,16 +33,14 @@
...
- * new score.
- * */
+ * Implements the rescoring logic. The top documents returned by solr with their original scores,
+ * will be processed by a {@link LTRScoringQuery} that will assign a new score to each document. The
+ * top documents will be resorted based on the new score.
+ */
 public class LTRRescorer extends Rescorer {
 
-  final private LTRScoringQuery scoringQuery;
+  private final LTRScoringQuery scoringQuery;
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java:43]: didn't realize this also did modifier ordering. happy to see it though.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java [hunk 33]
```diff
@@ -118,52 +112,63 @@ public TopDocs rescore(IndexSearcher searcher, TopDocs firstPassTopDocs,
...
-        } else {
-          // This subtraction can't overflow int
-          // because docIDs are >= 0:
-          return a.doc - b.doc;
-        }
-      }
-    });
+    Arrays.sort(
+        reranked,
+        new Comparator<ScoreDoc>() {
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java:132]: probably outside of scope here, but we could make a static instance of this comparator and reuse it.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java [hunk 34]
```diff
@@ -186,18 +191,40 @@ public void scoreFeatures(IndexSearcher indexSearcher,
         docBase = readerContext.docBase;
         scorer = modelWeight.scorer(readerContext);
       }
-      scoreSingleHit(indexSearcher, topN, modelWeight, docBase, hitUpto, hit, docID, scoringQuery, scorer, reranked);
+      scoreSingleHit(
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java:194]: ugh, these are both bad, not sure what I think we should do here.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java [hunk 35]
```diff
@@ -186,18 +191,40 @@ public void scoreFeatures(IndexSearcher indexSearcher,
...
+      int docBase,
+      int hitUpto,
+      ScoreDoc hit,
+      int docID,
+      LTRScoringQuery rerankingQuery,
+      LTRScoringQuery.ModelWeight.ModelScorer scorer,
+      ScoreDoc[] reranked)
+      throws IOException {
     final FeatureLogger featureLogger = rerankingQuery.getFeatureLogger();
     // Scorer for a LTRScoringQuery.ModelWeight should never be null since we always have to
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java:222]: Can we make a block comment here, I think that would collapse some of the line fragments better.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/OriginalScoreFeature.java [hunk 36]
```diff
@@ -28,48 +27,56 @@
...
+        SolrQueryRequest request,
+        Query originalQuery,
+        Map<String, String[]> efi)
+        throws IOException {
       super(OriginalScoreFeature.this, searcher, request, originalQuery, efi);
       w = searcher.createWeight(searcher.rewrite(originalQuery), ScoreMode.COMPLETE, 1);
-    };
-
+    }
+    ;
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/OriginalScoreFeature.java:79]: is this an extra semicolon?

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/interleaving/algorithms/TeamDraftInterleaving.java [hunk 37]
```diff
@@ -22,31 +22,31 @@
...
+ * ACM Press (2008) [4] O. Chapelle, T. Joachims, F. Radlinski, and Y. Yue. Large-scale validation
+ * and analysis of interleaved search evaluation. ACM TOIS, 30(1):1–41, Feb. (2012)
  */
 public class TeamDraftInterleaving implements Interleaving {
   public static Random RANDOM;
 
   static {
-    // We try to make things reproducible in the context of our tests by initializing the random instance
+    // We try to make things reproducible in the context of our tests by initializing the random
+    // instance
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/interleaving/algorithms/TeamDraftInterleaving.java:49]: combine with next line

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java [hunk 38]
```diff
@@ -113,74 +122,78 @@ public static LTRScoringModel getInstance(SolrResourceLoader solrResourceLoader,
...
+    this.allFeatures =
+        allFeatures != null ? Collections.unmodifiableList(new ArrayList<>(allFeatures)) : null;
     this.params = params != null ? Collections.unmodifiableMap(new LinkedHashMap<>(params)) : null;
     this.norms = norms != null ? Collections.unmodifiableList(new ArrayList<>(norms)) : null;
   }
 
   /**
-   * Validate that settings make sense and throws
-   * {@link ModelException} if they do not make sense.
+   * Validate that settings make sense and throws {@link ModelException} if they do not make sense.
```
Comment (madrob) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java:143]: surprised this didn't collapse into a single line

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java (hunk 33 above)
Comment (cpoerschke) [reply solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java:132]: > probably outside of scope here, but we could make a static instance of this comparator and reuse it.

https://issues.apache.org/jira/browse/SOLR-15496 created for it.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java (hunk 31 above)
Comment (cpoerschke) [reply solr/contrib/ltr/src/java/org/apache/solr/ltr/FeatureLogger.java]: > I don't like this

how about https://github.com/apache/solr/pull/189/commits/4d13b97e41c72381367c37a8799a21169c8e0fa3 as a replacement?

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java (hunk 34 above)
Comment (cpoerschke) [reply solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java:194]: hmm, yeah, ten arguments for the method is also quite a lot, perhaps a different method signature somehow with an intermediate structure or something. would need to give it some thought (beyond reformatting code that is)

Comment (cpoerschke) [reply solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java:194]: #192 would be one way to split the 10-args method.

**Comment by cpoerschke** [comment]
converting to draft until #192 and #228 are both merged into `main` and then into this branch.

**Comment by risdenk** [comment]
Hmmm I saw this after putting together https://github.com/apache/solr/pull/664 from SOLR-16029 - @cpoerschke do you plan to update this?

**Comment by cpoerschke** [comment]
> Hmmm I saw this after putting together #664 from [SOLR-16029](https://issues.apache.org/jira/browse/SOLR-16029) - @cpoerschke do you plan to update this?

Thanks @risdenk for the ping! Update done.

**Review by risdenk — APPROVED** [review]

**Comment by risdenk** [comment]
Thanks @cpoerschke!

**Comment by alessandrobenedetti** [comment]
I am currently away with limited connectivity, I'll check it later, in 2 weeks, but I see you merged already!

**Pull request merged**